packages/devtools/lib/src/timeline/timeline.dart

```diff
@@ -275,22 +295,29 @@ class TimelineScreen extends Screen {
     }
   }
 
-  /// Adds buttons to the timeline that will dump debug information to text
-  /// files and download them. This will only appear if either the
-  /// [debugTimeline] flag is true or the [debugCpuProfile] flag is true.
+  void _exportTimeline() {
+    // There will be a trailing comma on [debugTraceEvents]. Remove so that we
+    // do not have to deal with json formatting errors later.
+    final traceEvents = debugTraceEvents.isEmpty
+        ? ''
+        : debugTraceEvents.toString().replaceRange(
+            debugTraceEvents.length - 1, debugTraceEvents.length, '');
+    final String json = '{'
+        '"traceEvents":[$traceEvents],'
+        '"cpuProfile":[${debugCpuProfileResponse.toString()}]}';
+    downloadFile(json, 'timeline_${DateTime.now().toString()}.json');
```

Comment:
What does the output of `DateTime.now()` looks like? If it's not great, perhaps use `timeline_yyyy_mm_dd.json`?

Reply:
It looks like `timeline_2019-05-02 12_13_29.802`. I figured the timestamp would be helpful, as it is likely users will be exporting multiple profiles per day.

Reply:
I would avoid spaces here for sure. I don't know that we need the additional level of precision; if they have a file matching that on disk (for a previous export that day), the browser will likely offer to append a `-n` to the file?

We could also do `timeline_yyyy_mm_dd-ms.json`, where ms is the milliseconds since midnight. That you be a terse, unique file name.

Reply:
Switched to use timeline_yyyy_mm_dd-ms.json format:

i.e. `timeline_2019_5_2-1556825240984000.json`